pypeit/spectrographs/keck_hires.py

```diff
@@ -94,20 +262,297 @@ def check_frame_type(self, ftype, fitstbl, exprng=None):
+            return good_exp & (fitstbl['idname'] == 'Object')
         if ftype == 'bias':
             return good_exp & (fitstbl['idname'] == 'Bias')
         if ftype == 'dark':
             return good_exp & (fitstbl['idname'] == 'Dark')
         if ftype in ['pixelflat', 'trace']:
             # Flats and trace frames are typed together
-            return good_exp & ((fitstbl['idname'] == 'Flat') | (fitstbl['idname'] == 'IntFlat'))
+            return good_exp & (fitstbl['idname'] == 'IntFlat')
         if ftype in ['arc', 'tilt']:
+            # Arc and tilt frames are typed together
             return good_exp & (fitstbl['idname'] == 'Line')
 
         msgs.warn('Cannot determine if frames are of type {0}.'.format(ftype))
         return np.zeros(len(fitstbl), dtype=bool)
 
+
+    def get_rawimage(self, raw_file, det, spectrim=20):
+        """
+        Read raw images and generate a few other bits and pieces
+        that are key for image processing.
+
+        Based on readmhdufits.pro
+
+        Parameters
+        ----------
+        raw_file : :obj:`str`
+            File to read
+        det : :obj:`int`
+            1-indexed detector to read
+
+        Returns
+        -------
+        detector_par : :class:`pypeit.images.detector_container.DetectorContainer`
+            Detector metadata parameters.
+        raw_img : `numpy.ndarray`_
+            Raw image for this detector.
+        hdu : `astropy.io.fits.HDUList`_
+            Opened fits file
+        exptime : :obj:`float`
+            Exposure time read from the file header
+        rawdatasec_img : `numpy.ndarray`_
+            Data (Science) section of the detector as provided by setting the
+            (1-indexed) number of the amplifier used to read each detector
+            pixel. Pixels unassociated with any amplifier are set to 0.
+        oscansec_img : `numpy.ndarray`_
+            Overscan section of the detector as provided by setting the
+            (1-indexed) number of the amplifier used to read each detector
+            pixel. Pixels unassociated with any amplifier are set to 0.
+        """
+        # TODO -- Put a check in here to avoid data using the
+        #  original CCD (1 chip)
+
+
+        # Check for file; allow for extra .gz, etc. suffix
+        if not os.path.isfile(raw_file):
+            msgs.error(f'{raw_file} not found!')
+        hdu = io.fits_open(raw_file)
+
+        head0 = hdu[0].header
+
+        # Get post, pre-pix values
+        precol = head0['PRECOL']
+        postpix = head0['POSTPIX']
+        preline = head0['PRELINE']
+        postline = head0['POSTLINE']
+        detlsize = head0['DETLSIZE']
+        x0, x_npix, y0, y_npix = np.array(parse.load_sections(detlsize)).flatten()
+
+        # get the x and y binning factors...
+        #binning = head0['BINNING']
+
+        binning = self.get_meta_value(self.get_headarr(hdu), 'binning')
+#        # TODO: JFH I think this works fine
```

Comment:
remove?

Reply:
I think these comments can be removed when things actually work.